node-zerox/src/index.ts

```diff
@@ -47,13 +48,27 @@ export const zerox = async ({
   await fs.ensureDir(tempDirectory);
 
   // Download the PDF. Get file name.
-  const localPath = await downloadFile({ filePath, tempDir: tempDirectory });
+  const { localPath, response } = await downloadFile({
+    filePath,
+    tempDir: tempDirectory,
+  });
   if (!localPath) throw "Failed to save file to local drive";
 
-  const fileExtension = path.extname(localPath).toLowerCase();
+  let mimetype = response?.headers?.["content-type"];
+  if (!mimetype) {
+    mimetype = mime.lookup(localPath);
+  }
+
+  if (!mimetype) {
+    throw new Error("MIME type not found");
+  }
+
+  const fileExtension = mime.extension(mimetype) || "";
```

Comment:
Can we just use `mimetype` instead of `fileExtension`?

Reply:
I think we can just upload the below to be:
```
  if (mimetype !== "image/png") {
    let pdfPath: string;
    if (mimetype === "application/pdf") {
      pdfPath = localPath;
    } else {
      pdfPath = await convertFileToPdf({
        localPath,
        mimetype,
        tempDir: tempDirectory,
      });
    }
```

Reply:
sure I will change

Reply:
And then `convertFileToPdf` can be updated to allow for mimetypes (although we will have to add a default for the binary/octet-stream mimetype since it is not included in the mime-db package that mime-types uses):
```
export const convertFileToPdf = async ({
  localPath,
  mimetype,
  tempDir,
}: {
  localPath: string;
  mimetype: string;
  tempDir: string;
}): Promise<string> => {
  const inputBuffer = await fs.readFile(localPath);
  let extension = mime.extension(mimetype);
  if (!extension) {
    if (mimetype === "binary/octet-stream") {
      extension = "bin";
    } else {
      throw new Error("File extension missing");
    }
    throw new Error("File extension missing");
  }

```